crates/pop-contracts/src/up.rs

```diff
@@ -93,9 +102,95 @@ pub async fn set_up_upload(
 
 	let upload_exec: UploadExec<DefaultConfig, DefaultEnvironment, Keypair> =
 		UploadCommandBuilder::new(extrinsic_opts).done().await?;
+
 	Ok(upload_exec)
 }
 
+pub async fn get_upload_payload(up_opts: UpOpts) -> anyhow::Result<Vec<u8>> {
+	let code = get_contract_code(up_opts.path.as_ref()).await?;
+
+	let storage_deposit_limit: Option<u128> = None;
+	let upload_code = contract_extrinsics::extrinsic_calls::UploadCode::new(
+		code,
+		storage_deposit_limit,
+		contract_extrinsics::upload::Determinism::Enforced,
+	);
+
+	// TODO: review placement
+	let rpc_client = subxt::backend::rpc::RpcClient::from_url(up_opts.url.as_str()).await?;
+	let client = subxt::OnlineClient::<SubstrateConfig>::from_rpc_client(rpc_client).await?;
+
+	let call_data = upload_code.build();
+	let mut encoded_data = Vec::<u8>::new();
+	call_data.encode_call_data_to(&client.metadata(), &mut encoded_data)?;
```

Comment:
I'll take a look on replicate this behaviour in `pop-cli` without need of `contract-extrinsics = { git = "https://github.com/use-ink/cargo-contract", branch = "peter/chore-make-types-pub" }`